app/src/main/java/org/mozilla/fenix/collections/CollectionCreationController.kt

```diff
@@ -158,18 +158,27 @@ class DefaultCollectionCreationController(
     fun stepBack(
         backFromStep: SaveCollectionStep
     ): SaveCollectionStep? {
+        /*
+        Will return the next valid state according to this diagram.
+
+        Name Collection -> Select Collection -> Select Tabs -> (dismiss fragment) <- Rename Collection
+         */
+
+        val tabCollectionCount = store.state.tabCollections.size
+        val tabCount = store.state.tabs.size
+
         return when (backFromStep) {
-            SaveCollectionStep.SelectTabs, SaveCollectionStep.RenameCollection -> null
-            SaveCollectionStep.SelectCollection -> if (store.state.tabs.size <= 1) {
-                stepBack(SaveCollectionStep.SelectTabs)
+            SaveCollectionStep.NameCollection -> if (tabCollectionCount > 0) {
+                SaveCollectionStep.SelectCollection
```

Comment:
The diff here is a little hard to read.  This function's easier to understand by looking at 'files changed'.